x/ccv/child/module.go

```diff
@@ -249,6 +278,32 @@ func (am AppModule) OnChanOpenInit(
 		return err
 	}
 
+	///////////////////////////////////////////////////
+	// Initialize distribution token transfer channel
+	// NOTE the handshake for this channel is handled by
+	//      the ibc-go/transfer module
+
+	transferPortID := "transfer"
+	am.keeper.SetDistributionTransmissionPort(ctx, transferPortID)
+
+	distChan := channeltypes.Channel{
+		State:          channeltypes.UNINITIALIZED,
+		Ordering:       channeltypes.UNORDERED,
+		Counterparty:   counterparty,          // XXX is it correct to reuse
+		ConnectionHops: connectionHops,        // XXX is it correct to reused
+		Version:        transfertypes.Version, // XXX verify that this is correct
+	}
+	distrTransferMsg := channeltypes.MsgChannelOpenInit{
+		PortId:  transferPortID,
+		Channel: distChan,
+		Signer:  "", // signer unused
+	}
+	resp, err := am.keeper.ChannelOpenInit(ctx, &distrTransferMsg)
```

Comment:
This is a good point, cc: @rigelrozanski

Reply:
Okay I've moved the code to the Ack - a small issue is introduced now, we need to get the `connectionHops` from somewhere. Previously, when the handshake was initiated in `OnChanOpenInit` the connection hops was passed in through the function signature. There are two good options as I see it: 
 1) pass back connectionHops from provider OnChanOpenTry through version-metadata (easy). However this can only work if the connectionHops string is the same on the provider chain (is it? @AdityaSripal  @mpoke) 
 2)  store the connectionHops from OnChanOpenInit on the consumer chain then retrieve it during OnChanOpenAck (little bit more annoying because we need to setup a key/do all the store boilerplate code... but not a huge deal)

(side thought any good documentation for what the connection hops is kicking around?)

Reply:
https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#definitions
> The `connectionHops` stores the list of connection identifiers, in order, along which packets sent on this channel will travel. At the moment this list must be of length 1. In the future multi-hop channels may be supported.

I'd assume that the cleaner option is to store the connectionHops from OnChanOpenInit on the consumer chain. I'm not sure that the connectionHops on the provider will contain the same connections identifiers. 

@AdityaSripal  How is the relayer creating `MsgChannelOpenInit`? We can do it in the same way since we already know the connection identifier.

Reply:
connection hops are saved in the channel. Resolved